File: codefights/elixir/phone_call/code.exs
def phoneCall(min1, min2_10, min11, s) do
    calculate_cost(min1, min2_10, min11, s)
end

def calculate_cost(min1, min2_10, min11, s) when (s > min1) do
    cond do 
        ((s - min1) - min2_10*9) > 0 -> 10 +  elem((((s - min1) - min2_10*9)/min11) |> to_string |> Integer.parse(10), 0)
        ((s - min1) - min2_10*9) < 0 -> 1 + elem(((s - min1)/min2_10) |> to_string |> Integer.parse(10), 0)
    end
end

def calculate_cost(min1, min2_10, min11, s) when (s < min1) do
    0
end

def calculate_cost(min1, min2_10, min11, s) when (s == min1) do
    1
end


# 1. Minus cost of 1st Minute from s
# 2. Take the total and minus from minute 2 to 10: (s - min1) - (s - min2_10*9)
# 3. If s is still not zero: remaining / min11 

# Total => 1 + 9 + total from step 3

File: codefights/kotlin/phone_call/code.kt
fun phoneCall(min1: Int, min2_10: Int, min11: Int, s: Int): Int {
    return when {
        (s == min1) -> 1
        (s < min1) -> 0
        ((s - min1) - min2_10*9) > 0 -> 10 + ((s - min1) - min2_10*9)/min11
        ((s - min1) - min2_10*9) < 0 -> 1 + (s - min1)/min2_10
        else -> 0
    }
}

